packages/toc/src/generators/markdown/options_manager.ts

```diff
@@ -66,6 +72,18 @@ class OptionsManager extends Registry.IOptionsManager {
     return this._numbering;
   }
 
+  /**
+   * Gets/sets ToC generator numbering h1 headers.
+   */
+  set numberingH1(value: boolean) {
+    this._numberingH1 = value;
+    this._widget.update();
```

Comment:
Please condition the update on a change

```suggestion
    if (this._numberingH1 !== value) {
      this._numberingH1 = value;
      this._widget.update();
    }
```